### PR TITLE
Move Parquet related classes to the new package

### PR DESCRIPTION
**Pull request by Jackie-Jiang**
Create a new package: pinot-parquet for all Parquet related classes
Remove the dependency of parquet-avro from pinot-common

### CODE REVIEW

File: pinot-core/src/main/java/org/apache/pinot/core/data/readers/RecordReaderFactory.java [hunk 1]
```diff
@@ -46,10 +46,10 @@ public static RecordReader getRecordReader(SegmentGeneratorConfig segmentGenerat
     // If this is set, this will override the file format
     if (recordReaderPath != null) {
       if (fileFormat != FileFormat.OTHER) {
-        // We currently have default file format set to AVRO inside segment generator config,
-        // do not want to break this behavior for clients.
-        LOGGER.warn("Using recordReaderPath {} to read segment, ignoring fileformat {}",
-            recordReaderPath, fileFormat.toString());
+        // NOTE: we currently have default file format set to AVRO inside segment generator config, do not want to break
+        // this behavior for clients.
+        LOGGER
```
Comment (snleee) [inline pinot-core/src/main/java/org/apache/pinot/core/data/readers/RecordReaderFactory.java]: Previous format looks better to me :)

File: pinot-hadoop/src/main/java/org/apache/pinot/hadoop/job/mapper/SegmentCreationMapper.java [hunk 2]
```diff
@@ -101,8 +101,7 @@ public void setup(Context context)
     if (readerConfigFile != null) {
       _readerConfigFile = new Path(readerConfigFile);
     }
-
-    _recordReaderPath = _jobConf.get(JobConfigConstants.RECORD_READER_PATH, null);
+    _recordReaderPath = _jobConf.get(JobConfigConstants.RECORD_READER_PATH);
```
Comment (snleee) [inline pinot-hadoop/src/main/java/org/apache/pinot/hadoop/job/mapper/SegmentCreationMapper.java]: `RECORD_READER_CLASS_NAME` seems to be more correct name... But, I guess that we are already using this config?

**Comment by snleee** [comment]
TODO: Let's add the documentation on adding a new record reader and link orc/parquet prs as examples.

File: pinot-parquet/src/main/java/org/apache/pinot/parquet/data/readers/ParquetUtils.java [hunk 3]
```diff
@@ -26,67 +28,53 @@
...
-   * @return a ParquetReader
-   * @throws IOException
+   * Returns a ParquetReader with the given path.
    */
-  public static ParquetReader<GenericRecord> getParquetReader(String fileName)
+  public static ParquetReader<GenericRecord> getParquetReader(Path path)
       throws IOException {
-    Path dataFsPath = new Path(fileName);
-    return AvroParquetReader.<GenericRecord>builder(dataFsPath).disableCompatibility().withDataModel(GenericData.get())
+    //noinspection unchecked
```
[the remark this answers is not in the document]
Comment (Jackie-Jiang) [reply pinot-parquet/src/main/java/org/apache/pinot/parquet/data/readers/ParquetUtils.java]: IDE can read this and suppress the warning for this cast.

File: pinot-hadoop/src/main/java/org/apache/pinot/hadoop/job/mapper/SegmentCreationMapper.java (hunk 2 above)
Comment (Jackie-Jiang) [reply pinot-hadoop/src/main/java/org/apache/pinot/hadoop/job/mapper/SegmentCreationMapper.java]: Yes, so cannot change it lol

File: pinot-core/src/main/java/org/apache/pinot/core/data/readers/RecordReaderFactory.java (hunk 1 above)
Comment (Jackie-Jiang) [reply pinot-core/src/main/java/org/apache/pinot/core/data/readers/RecordReaderFactory.java]: I agree, but just follow the auto-format...

**Pull request merged**